client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -204,13 +202,19 @@ private NettyAdapter(int eventLoopGroupSize, int channelPoolSize) {
             }).onErrorResumeNext((Throwable throwable) -> {
                 if (throwable instanceof EncoderException) {
                     LoggerFactory.getLogger(getClass()).warn("Got EncoderException: " + throwable.getMessage());
+                    //TODO what is this, a retry? Should have a time delay? Max number of retries?
```

Comment:
I've been getting EncoderException under load in cases where the server has responded with 503 or otherwise failed to receive the request. As far as I can tell it relates to the internal state of the Netty HttpRequestEncoder getting corrupted. I've been resolving it by throwing away the connection/pipeline and grabbing a new one.

There are probably some conditions (i.e. the early response from server case that seems to keep coming up) where  we should just be closing the channel and discarding the related HttpRequestEncoder but we're failing to do so.

Reply:
Weird eh. As long as you are happy we don't end up in some high cpu infinite loop because we don't inject a time delay...

Reply:
I'd like most of all to root out what's putting the encoder into a bad state when the server sends an error response early, and then remove the usage of `onErrorResumeNext` here. If it can't be solved in a reasonable amount of time I will just add a delay on each retry. Thanks for pointing this one out.